### PR TITLE
Mantine Card

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/32268
Context: https://metaboat.slack.com/archives/C057WD5L0JG/p1693935868971129

How to verify:
- `yarn storybook`
- Open http://localhost:6006/?path=/docs/data-display-card--default-story
- Make sure `Paper` properties can be configured (`radius` etc)

Screenshot 2023-09-26 at 10 28 46

### CODE REVIEW

**Comment by oisincoveney** [comment]
@ranquild I think `Accordion` is actually being used in `frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx`

File: frontend/src/metabase/ui/components/data-display/Card/Card.styled.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { MantineTheme, MantineThemeOverride } from "@mantine/core";
+
+export const getCardOverrides = (): MantineThemeOverride["components"] => ({
```
Comment (WiNloSt) [inline frontend/src/metabase/ui/components/data-display/Card/Card.styled.tsx:3]: How do you determine what style should be overridden?

**Review by WiNloSt — COMMENTED** [review]
I'm not sure how to review this PR. I checked the [EPIC](https://github.com/metabase/metabase/issues/32268), but couldn't find any thing regarding the card component or any Figma link at all.

File: frontend/src/metabase/ui/components/data-display/Card/Card.styled.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/ui/components/data-display/Card/Card.styled.tsx:3]: Initially we thought that no styles need to be changed. But by looking at the implementation they use custom border colors that we need to override.

**Comment by ranquild** [comment]
> I'm not sure how to review this PR. I checked the [EPIC](https://github.com/metabase/metabase/issues/32268), but couldn't find any thing regarding the card component or any Figma link at all.

@WiNloSt https://metaboat.slack.com/archives/C057WD5L0JG/p1693935868971129

**Review by WiNloSt — APPROVED** [review]
Looks simple. There's very minimal overridden styles, otherwise, it uses all the base from Mantine.

**Review by mazameli — APPROVED** [review]

**Pull request merged**